packages/core/strapi/lib/core-api/service/index.d.ts

```diff
@@ -20,3 +20,8 @@ export interface CollectionTypeService extends BaseService {
 
 export type Service = SingleTypeService | CollectionTypeService;
 
+export type GenericService = Partial<Service> & {
+  [method: string | number | symbol]: (...args:unknown) => Promise<Entity> | Entity;
```

Comment:
It won't necessarily return an Entity here :slightly_smiling_face:

Reply:
hmmmm, should we have it be `any` or `unknown` ?

Reply:
Can use a generic here:
`[method: string | number | symbol]: <T = unknown>(...args:unknown) => T;`

Reply:
@stafyniaksacha agreed probably using any would be better for now, will go with your suggestion